app/src/components/DataRequirements.tsx

```diff
@@ -0,0 +1,142 @@
+import { Accordion, Center, createStyles, em, getBreakpointValue, List, Paper, rem } from '@mantine/core';
+
+interface CodeInterface {
+  code: string | undefined;
+  system: string | undefined;
+}
+
+const useStyles = createStyles(theme => ({
+  card: {
+    borderRadius: 12,
+    border: `${rem(2)} solid ${theme.colors.gray[3]}`,
+    width: '1200px',
+    [`@media (max-width: ${em(getBreakpointValue(theme.breakpoints.lg) - 1)})`]: {
+      width: '100%'
+    }
+  }
+}));
+
+/**
+ * Component which displays all data requirements of a specified resource and displays them
+ * as resource cards that contain all required information
+ */
+function DataRequirements({ type, extension, dateFilter, codeFilter }: fhir4.DataRequirement) {
+  const { classes } = useStyles();
+  const codes: CodeInterface[] = [];
+
+  const codeFiltersWithVS = codeFilter?.filter(cf => {
+    return cf?.valueSet !== undefined;
+  });
```

Comment:
```suggestion
  const codeFiltersWithVS = codeFilter?.filter(cf => cf?.valueSet);
```
You can shorten this a bit!